src/toolbox.js

```diff
@@ -61,6 +64,10 @@ var Toolbox = GObject.registerClass({
         this._buttonReset.connect('clicked', this._onResetClicked.bind(this));
         this._buttonMinimize.connect('clicked', this._onMinimize.bind(this));
         this.setBusy(false);
+
+        this._gameState = GameState.getDefault();
+        this._gameState.connect('g-signal', this._onStateChanged.bind(this));
```

Comment:
If you use the (unfortunately JS-only) signal that the proxy object creates, then you can do the state change without another sync call to get the state.

See https://github.com/endlessm/hack-hackdex/blob/master/com.endlessm.Hackdex_chapter_one/controller/meshDev.js#L41 for an example

Reply:
I can do the same thing with the current approach :) But since I _have_ to read it while constructing the toolbox, and whole try/unpack boilerplate hurts, is just looked simpler to do it this way. But I can surely save that one call.